core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -100,8 +101,8 @@ object ConsoleConsumer extends Logging {
         consumer.receive()
       } catch {
         case e: Throwable => {
-          error("Error processing message, stopping consumer: ", e)
-          consumer.stop()
+          error("Error processing message, terminating consumer process: ", e)
+          // Consumer will be stopped in shutdown hook
```

Comment:
@rajinisivaram I think KafkaConsumer.close() should not throw ```ConsumerWakeupException``` ever since it should only be thrown in poll(); if it does, that should be a bug.

There are some discrepancies between old and new consumer regarding ```consumer.cleanup()``` and ```consumer.stop()```: for the old consumer cleanup seems done before stop although it does not matter since they are actually triggering the same function, but for new consumer it seems cleanup should be done after stop. 

Maybe we can remove the ```consumer.cleanup()``` interface in BaseConsumer, and let the new consumer implement ```consumer.stop()``` as a wakeup called followed by close.